### PR TITLE
Deployments confusion

### PR DESCRIPTION
**Pull request by rubbish**
This clears up `Deployments` and `DeploymentIDs` on the application. `Deployments` now matches what the JSON says and `app.DeploymentIDs()` is a nice wrapper around that.

This one builds on #71 as otherwise they'll conflict.

### CODE REVIEW

**Comment by eicca** [comment]
Should the interface be changed as well: https://github.com/Banno/go-marathon/blob/deployments-confusion/client.go#L107?

File: application.go [hunk 1]
```diff
@@ -191,13 +191,13 @@ func (r *Application) HasHealthChecks() bool {
 }
 
 // Deployments retrieves the application deployments ID
```
Comment (eicca) [inline application.go]: The comment won't pass `golint` now. Can you change it please to `DeploymentIDs does something`?

**Comment by rubbish** [comment]
Good call. I'll get those changed up.

**Comment by rubbish** [comment]
@eicca I don't believe the client or marathon interface has to change. This is only on the Application interface.

**Comment by eicca** [comment]
@rubbish I'm sorry, I got confused by the methods with the same name :)

**Pull request merged**